### PR TITLE
Update config to use filter jar paths and load jars

### PR DESCRIPTION
**Pull request by shashank11p**
Pending: update submodule after https://github.com/hypertrace/agent-config/pull/47 is merged

### CODE REVIEW

File: javaagent-core/src/test/resources/config.yaml [hunk 1]
```diff
@@ -14,3 +14,7 @@ dataCapture:
     request: true
     response: false
   httpBody:
+javaagent:
+  filter_jar_paths:
+    - /path1.jar
+    - /path/2/jar.jar
```
Comment (pavolloffay) [inline javaagent-core/src/test/resources/config.yaml:20]: add new line break

File: javaagent-core/src/test/java/org/hypertrace/agent/core/config/HypertraceConfigTest.java [hunk 2]
```diff
@@ -111,6 +113,17 @@ private void assertConfig(AgentConfig agentConfig) {
         true, agentConfig.getDataCapture().getHttpBody().getRequest().getValue());
     Assertions.assertEquals(
         true, agentConfig.getDataCapture().getRpcBody().getRequest().getValue());
+    Assertions.assertEquals(2, agentConfig.getJavaagent().getFilterJarPathsCount());
+    Assertions.assertTrue(
```
Comment (pavolloffay) [inline javaagent-core/src/test/java/org/hypertrace/agent/core/config/HypertraceConfigTest.java:117]: Make these last two asserts more specific and test concrete indices

File: javaagent-core/src/main/java/org/hypertrace/agent/core/config/HypertraceConfig.java [hunk 3]
```diff
@@ -174,6 +175,9 @@ static AgentConfig load(String filename) throws IOException {
     if (configBuilder.getPropagationFormatsList().isEmpty()) {
       configBuilder.addPropagationFormats(PropagationFormat.TRACECONTEXT);
     }
+    if (!configBuilder.hasJavaagent()) {
+      configBuilder.setJavaagent(JavaAgent.newBuilder());
```
Comment (pavolloffay) [inline javaagent-core/src/main/java/org/hypertrace/agent/core/config/HypertraceConfig.java:179]: why do we need this?

File: filter-api/src/main/java/org/hypertrace/agent/filter/FilterRegistry.java [hunk 4]
```diff
@@ -70,6 +78,27 @@ private static Filter load() {
     return new MultiFilter(filters);
   }
 
+  private static ClassLoader loadJars() {
+    List<StringValue> jarPaths = HypertraceConfig.get().getJavaagent().getFilterJarPathsList();
+    List<File> files = new ArrayList<>();
+    jarPaths.forEach(
+        jarPath -> {
+          File file = new File(jarPath.getValue());
+          System.out.println(jarPath.getValue());
```
Comment (pavolloffay) [inline filter-api/src/main/java/org/hypertrace/agent/filter/FilterRegistry.java:87]: remove

File: filter-api/src/main/java/org/hypertrace/agent/filter/FilterRegistry.java [hunk 5]
```diff
@@ -70,6 +78,27 @@ private static Filter load() {
...
+          files.add(file);
+        });
+    int i = 0;
```
Comment (pavolloffay) [inline filter-api/src/main/java/org/hypertrace/agent/filter/FilterRegistry.java:90]: remove

File: filter-api/src/main/java/org/hypertrace/agent/filter/FilterRegistry.java [hunk 6]
```diff
@@ -70,6 +78,27 @@ private static Filter load() {
...
+    URL[] urls = new URL[jarPaths.size()];
+    for (File file : files) {
+      try {
+        urls[i] = file.toURI().toURL();
+      } catch (MalformedURLException e) {
```
Comment (pavolloffay) [inline filter-api/src/main/java/org/hypertrace/agent/filter/FilterRegistry.java:95]: log the exception as well

File: filter-api/src/main/java/org/hypertrace/agent/filter/FilterRegistry.java [hunk 7]
```diff
@@ -70,6 +78,27 @@ private static Filter load() {
     return new MultiFilter(filters);
   }
 
+  private static ClassLoader loadJars() {
+    List<StringValue> jarPaths = HypertraceConfig.get().getJavaagent().getFilterJarPathsList();
+    List<File> files = new ArrayList<>();
```
Comment (pavolloffay) [inline filter-api/src/main/java/org/hypertrace/agent/filter/FilterRegistry.java:83]: Do we need to create file objects? It does not seem to be needed. We can create directly URLs

File: filter-api/build.gradle.kts [hunk 8]
```diff
@@ -9,5 +9,6 @@ dependencies {
    api("io.opentelemetry:opentelemetry-api:${versions["opentelemetry"]}")
    api(project(":javaagent-core"))
    implementation("com.google.auto.service:auto-service:1.0-rc7")
+   implementation("org.slf4j:slf4j-api:1.7.30")
```
Comment (pavolloffay) [inline filter-api/build.gradle.kts:12]: It seems we should define SLF4J in the versions map and reuse it

File: testing-common/build.gradle.kts [hunk 9]
```diff
@@ -20,9 +20,9 @@ dependencies {
     implementation("io.opentelemetry.javaagent:opentelemetry-javaagent-spi:${versions["opentelemetry_java_agent"]}")
     implementation("io.opentelemetry.instrumentation:opentelemetry-instrumentation-api:${versions["opentelemetry_java_agent"]}")
     implementation("ch.qos.logback:logback-classic:1.2.3")
-    implementation("org.slf4j:log4j-over-slf4j:1.7.30")
-    implementation("org.slf4j:jcl-over-slf4j:1.7.30")
-    implementation("org.slf4j:jul-to-slf4j:1.7.30")
+    implementation("org.slf4j:log4j-over-slf4j:${versions["slf4j"]}")
```
Comment (pavolloffay) [inline testing-common/build.gradle.kts:23]: ❤️  nice work :)

File: filter-api/src/main/java/org/hypertrace/agent/filter/FilterRegistry.java [hunk 10]
```diff
@@ -70,6 +77,23 @@ private static Filter load() {
     return new MultiFilter(filters);
   }
 
+  private static ClassLoader loadJars() {
+    List<StringValue> jarPaths = HypertraceConfig.get().getJavaagent().getFilterJarPathsList();
+    URL[] urls = new URL[jarPaths.size()];
+    int i = 0;
+    for (StringValue jarPath : jarPaths) {
+      try {
+        URL url = new URL("file", "", -1, jarPath.getValue());
```
Comment (pavolloffay) [inline filter-api/src/main/java/org/hypertrace/agent/filter/FilterRegistry.java:86]: Do we know that this works on Windows?

Comment (shashank11p) [reply filter-api/src/main/java/org/hypertrace/agent/filter/FilterRegistry.java:86]: No.. I am not sure

**Review by pavolloffay — APPROVED** [review]

**Pull request merged**